test/parallel/test-process-load-env-file.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+const common = require('../common');
+const fixtures = require('../../test/common/fixtures');
+const path = require('node:path');
+const assert = require('node:assert');
+const { describe, it } = require('node:test');
+
+const validEnvFilePath = fixtures.path('dotenv/valid.env');
+const missingEnvFile = fixtures.path('dotenv/non-existent-file.env');
+
+describe('process.loadEnvFile()', () => {
+
+  it('supports passing path', async () => {
+    const code = `
+      process.loadEnvFile(${JSON.stringify(validEnvFilePath)});
+      const assert = require('assert');
+      assert.strictEqual(process.env.BASIC, 'basic');
+    `.trim();
+    const child = await common.spawnPromisified(
+      process.execPath,
+      [ '--eval', code ],
+      { cwd: __dirname },
+    );
+    assert.strictEqual(child.stderr, '');
+    assert.strictEqual(child.code, 0);
+  });
+
+  it('supports not-passing a path', async () => {
+    // Uses `../fixtures/dotenv/.env` file.
+    const code = `
+      process.loadEnvFile();
+      const assert = require('assert');
+      assert.strictEqual(process.env.BASIC, 'basic');
+    `.trim();
+    const child = await common.spawnPromisified(
+      process.execPath,
+      [ '--eval', code ],
+      { cwd: path.join(__dirname, '../fixtures/dotenv') },
+    );
+    assert.strictEqual(child.stderr, '');
+    assert.strictEqual(child.code, 0);
+  });
+
+  it('should throw when file does not exist', async () => {
+    assert.throws(() => {
+      process.loadEnvFile(missingEnvFile);
+    }, {
+      code: 'ENOENT',
+      errno: -2,
+    });
+  });
```

Comment:
```suggestion
  });
  
  it('should throw when `.env` does not exist', async () => {
    const child = await common.spawnPromisified(
      process.execPath,
      [ '--eval', 'assert.throws(() => process.loadEnvFile(), { code: "ENOENT" })'],
      { cwd: fixtures.path() },
    );
    assert.strictEqual(child.stderr, '');
    assert.strictEqual(child.stdout, '');
    assert.strictEqual(child.code, 0);
  });
```

Reply:
Why does this need to be in a child process? Wouldn’t it be faster if it wasn’t?

Reply:
Child process is actually 12 times slower than the current way.

Reply:
What I'm asking is to validate that the version with no arguments throws the correct error. I don't really care if we use a child process or not for that.

Reply:
Ah yes! I missed that. Sorry. Updating the PR now.